### PR TITLE
Added FakePlayer module

### PR DESCRIPTION
**Pull request by ciruu1**
Added FakePlayer module to try different pvp settings with a ghost player.

### CODE REVIEW

**Comment by dominikaaaa** [comment]
This looks pasted ngl. Also, maybe make the UUID / name customizable

**Comment by ciruu1** [comment]
> This looks pasted ngl. Also, maybe make the UUID / name customizable

Hey, this is the code that i wrote by myself, there's no other way to code that, if you found another way tell me then. And okay i can make the name/skin customizable, i'll update the request later.

**Comment by dominikaaaa** [comment]
I mean, there is, seeing as mine is written differently. If you really did write it, then your display name is wrong.

**Comment by ciruu1** [comment]
> I mean, there is, seeing as mine is written differently. If you really did write it, then your display name is wrong.

I didn't want to put my name as display name, that's why i put Ihacked's name, but as you said i'll make it customizable.

**Comment by dominikaaaa** [comment]
I meant the display name that you put doesn't match the uuid

**Comment by ciruu1** [comment]
> I meant the display name that you put doesn't match the uuid

Oh that's why i choose Ihacked's uuid, because somebody could steal his name. I'm working on make it customizable, i should have it soon.

**Comment by ionar2** [comment]
Call the super functions in onEnable and onDisable and fix the spaces, then probably looks good

**Comment by ciruu1** [comment]
> Call the super functions in onEnable and onDisable and fix the spaces, then probably looks good

Sure, i'm working right now on spawning a custom player, once i finish that i'll reformat all code and then update the request

**Comment by ciruu1** [comment]
Okay it works with only some names, let me fix it to work it with all names

**Comment by dominikaaaa** [comment]
I converted this to a draft, let us know when it's ready to merge c:

**Comment by ciruu1** [comment]
> I converted this to a draft, let us know when it's ready to merge c:

Okay, i'm not sure if i can put custom skins on entities but if i can't do it i'll make the last update and then i think that is ready

**Comment by ionar2** [comment]
Also you should add an override for toggleNoSave so this doesn't load as enabled on client restart

File: src/main/java/me/ionar/salhack/module/misc/FakePlayer.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+import org.apache.commons.io.IOUtils;
+
+public class FakePlayer extends Module {
+    public FakePlayer() {
+        super("FakePlayer", new String[]
+                { "Fake" }, "Summons a fake player", "NONE", 0xDADB25, ModuleType.MISC);
+    }
+
+    public final Value<String> name = new Value<>("Name", new String[] {"name"}, "Name of the fake player", "IHackedXVIDEOs");
```
Comment (ionar2) [inline src/main/java/me/ionar/salhack/module/misc/FakePlayer.java:23]: I rather this be jared2013

Comment (ciruu1) [reply src/main/java/me/ionar/salhack/module/misc/FakePlayer.java:23]: Okay, i'll change it

**Comment by ciruu1** [comment]
Ok tell me what do you think about it, in a future i could try to put a custom skin on the character but it's too much work for this moment

**Comment by ionar2** [comment]
That's fine, we can improve it later

File: src/main/java/me/ionar/salhack/module/misc/FakePlayer.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+                { "Fake" }, "Summons a fake player", "NONE", 0xDADB25, ModuleType.MISC);
+    }
+
+    public static final Value<String> name = new Value<>("Name", new String[] {"name"}, "Name of the fake player", "jared2013");
+
+    private EntityOtherPlayerMP fakePlayer;
+
+    @Override
+    public void onEnable() {
+        super.onEnable();
```
Comment (ionar2) [inline src/main/java/me/ionar/salhack/module/misc/FakePlayer.java:27]: fakePlayer should be set to null here again

File: src/main/java/me/ionar/salhack/module/misc/FakePlayer.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        if (mc.world == null)
+            return;
```
Comment (ionar2) [inline src/main/java/me/ionar/salhack/module/misc/FakePlayer.java:29]: toggle and return here.

File: src/main/java/me/ionar/salhack/module/misc/FakePlayer.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+import org.apache.commons.io.IOUtils;
+
+public class FakePlayer extends Module {
+    public FakePlayer() {
+        super("FakePlayer", new String[]
+                { "Fake" }, "Summons a fake player", "NONE", 0xDADB25, ModuleType.MISC);
+    }
+
+    public static final Value<String> name = new Value<>("Name", new String[] {"name"}, "Name of the fake player", "jared2013");
```
Comment (ionar2) [inline src/main/java/me/ionar/salhack/module/misc/FakePlayer.java:21]: options should be above the constructor

File: src/main/java/me/ionar/salhack/module/misc/FakePlayer.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    public void toggleNoSave(){
+
+    }
+
+    //Getting uuid from a name
+    public static String getUuid(String name) {
+        JsonParser parser = new JsonParser();
+        String url = "https://api.mojang.com/users/profiles/minecraft/" + name;
+        try {
+            @SuppressWarnings("deprecation")
```
Comment (ionar2) [inline src/main/java/me/ionar/salhack/module/misc/FakePlayer.java:63]: Could you try and not use deprecated functions? If you can't figure this one out, it's fine, we can fix it in the future

File: src/main/java/me/ionar/salhack/module/misc/FakePlayer.java [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+            JsonObject UUIDObject = (JsonObject) parser.parse(UUIDJson);
+            return reformatUuid(UUIDObject.get("id").toString());
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return "error";
+    }
+
+    //Reformating a short uuid type into a long uuid type
+    private static String reformatUuid(String uuid){
```
Comment (ionar2) [inline src/main/java/me/ionar/salhack/module/misc/FakePlayer.java:75]: This function seems really hacky, but is probably fine.

**Comment by ciruu1** [comment]
Sure, i'll change it, probably will take some time to change the deprecated function because i need to search what can i use lol but i'll have it by today

**Comment by ciruu1** [comment]
Nevermind, only needed to specify the string encoding, i'm uploading it

File: src/main/java/me/ionar/salhack/module/misc/FakePlayer.java [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+
+    private EntityOtherPlayerMP fakePlayer;
+
+    @Override
+    public void onEnable() {
+        super.onEnable();
+        fakePlayer = null;
+
+        if (mc.world == null) {
+            ModuleManager.Get().GetMod(FakePlayer.class).toggle();
```
Comment (ionar2) [inline src/main/java/me/ionar/salhack/module/misc/FakePlayer.java:35]: You're already in the FakePlayer object, just write this.toggle();

Comment (ciruu1) [reply src/main/java/me/ionar/salhack/module/misc/FakePlayer.java]: Done

**Comment by ionar2** [comment]
Thank you for your PR :)

**Comment by ciruu1** [comment]
Thank you too :D

**Pull request merged**